src/dc_receive_imf.rs

```diff
@@ -181,7 +211,8 @@ pub(crate) async fn dc_receive_imf_inner(
         &mut sent_timestamp,
         from_id,
         &mut hidden,
-        seen,
+        seen || replace_partial_download,
```

Comment:
Why this change? The variable is still named `seen` in `add_parts()`.

Reply:
`seen` comes from the server and may be incorrect for whatever reason. replace_partial_download is a much clearer signal here to not set the state wrong.

however, maybe we should better already do this check on calling `dc_receive_imf`.